File: mcp_client.ts
import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";
import { CallToolResult } from "@modelcontextprotocol/sdk/types.js";
import * as readline from 'node:readline/promises';
import { stdin as input, stdout as output } from 'node:process';

async function main() {
    // Create transport to communicate with Python MCP server
    const transport = new StdioClientTransport({
        command: "python",
        args: ["mcp_server.py"]
    });

    // Create MCP client
    const client = new Client(
        {
            name: "string-reverser-client",
            version: "1.0.0"
        },
        {
            capabilities: {
                tools: {}
            }
        }
    );

    // Connect to the server
    await client.connect(transport);
    console.log("Connected to MCP server");

    // Create readline interface
    const rl = readline.createInterface({ input, output });

    // Get input from user
    const text = await rl.question('Enter text to reverse: ');
    rl.close();

    // Call the reverse_string tool
    const result = await client.callTool({
        name: "reverse_string",
        arguments: { text }
    }) as CallToolResult;

    // Print the result
    console.log(`Reversed text: ${result.content[0].text}`);
}

main().catch(console.error); 

File: mcp_client.py
from mcp import ClientSession, StdioServerParameters, types
from mcp.client.stdio import stdio_client
import asyncio
from pdb import set_trace

async def main():
    # Create server parameters for stdio connection
    server_params = StdioServerParameters(
        command="python",
        args=["mcp_server.py"]
    )

    async with stdio_client(server_params) as (read, write):
        async with ClientSession(read, write) as session:
            # Initialize the connection
            await session.initialize()
            print("Connected to MCP server")

            # Get input from user
            text = input("Enter text to reverse: ")

            # Call the reverse_string tool
            result = await session.call_tool(
                "reverse_string",
                arguments={"text": text}
            )

            # Print the result - accessing as object properties
            reversed_text = result.content[0].text

            print(f"Reversed text: {reversed_text}")

            print(50*"+", "DONE", 50*"+")
            print(dir(session))
            result = await session.list_tools()
            print("/n/n/n", result.tools)
            print(50*"+", "DONE", 50*"+")
            
            

if __name__ == "__main__":
    asyncio.run(main()) 